### PR TITLE
Obsolete Rfc2898DeriveBytes.CryptDeriveKey

### PR DESCRIPTION
**Pull request by vcsjones**
Contributes to #50528.

### CODE REVIEW

**Comment by vcsjones** [comment]
Skipping API Review and setting for 6.0 per https://github.com/dotnet/runtime/issues/50528#issuecomment-893981312.

**Comment by vcsjones** [comment]
Test failures are https://github.com/dotnet/runtime/issues/56926 and https://github.com/dotnet/runtime/issues/56894.

File: docs/project/list-of-diagnostics.md [hunk 1]
```diff
@@ -87,6 +87,7 @@ The PR that reveals the implementation of the `<IncludeInternalObsoleteAttribute
 |  __`SYSLIB0030`__ | HMACSHA1 always uses the algorithm implementation provided by the platform. Use a constructor without the useManagedSha1 parameter. |
 |  __`SYSLIB0031`__ | EncodeOID is obsolete. Use the ASN.1 functionality provided in System.Formats.Asn1. |
 |  __`SYSLIB0032`__ | Recovery from corrupted process state exceptions is not supported; HandleProcessCorruptedStateExceptionsAttribute is ignored. |
+|  __`SYSLIB0033`__ | Rfc2898DeriveBytes.CryptDeriveKey is obsolete and no longer supported. Use PasswordDeriveBytes.CryptDeriveKey instead. |
```
Comment (jeffhandley) [inline docs/project/list-of-diagnostics.md:90]: Nit on the message (I just replaced "no longer" with "is not" in all of the other messages that used that phrase).

```suggestion
|  __`SYSLIB0033`__ | Rfc2898DeriveBytes.CryptDeriveKey is obsolete and is not supported. Use PasswordDeriveBytes.CryptDeriveKey instead. |
```

File: src/libraries/Common/src/System/Obsoletions.cs [hunk 2]
```diff
@@ -107,5 +107,8 @@ internal static class Obsoletions
 
         internal const string CorruptedStateRecoveryMessage = "Recovery from corrupted process state exceptions is not supported; HandleProcessCorruptedStateExceptionsAttribute is ignored.";
         internal const string CorruptedStateRecoveryDiagId = "SYSLIB0032";
+
+        internal const string Rfc2898CryptDeriveKeyMessage = "Rfc2898DeriveBytes.CryptDeriveKey is obsolete and no longer supported. Use PasswordDeriveBytes.CryptDeriveKey instead.";
```
Comment (jeffhandley) [inline src/libraries/Common/src/System/Obsoletions.cs:111]:
```suggestion
        internal const string Rfc2898CryptDeriveKeyMessage = "Rfc2898DeriveBytes.CryptDeriveKey is obsolete and is not supported. Use PasswordDeriveBytes.CryptDeriveKey instead.";
```

File: src/libraries/System.Security.Cryptography.Algorithms/ref/System.Security.Cryptography.Algorithms.cs [hunk 3]
```diff
@@ -645,6 +645,7 @@ public partial class Rfc2898DeriveBytes : System.Security.Cryptography.DeriveByt
         public System.Security.Cryptography.HashAlgorithmName HashAlgorithm { get { throw null; } }
         public int IterationCount { get { throw null; } set { } }
         public byte[] Salt { get { throw null; } set { } }
+        [System.ObsoleteAttribute("Rfc2898DeriveBytes.CryptDeriveKey is obsolete and no longer supported. Use PasswordDeriveBytes.CryptDeriveKey instead.", DiagnosticId = "SYSLIB0033", UrlFormat = "https://aka.ms/dotnet-warnings/{0}")]
```
Comment (jeffhandley) [inline src/libraries/System.Security.Cryptography.Algorithms/ref/System.Security.Cryptography.Algorithms.cs:648]:
```suggestion
        [System.ObsoleteAttribute("Rfc2898DeriveBytes.CryptDeriveKey is obsolete and is not supported. Use PasswordDeriveBytes.CryptDeriveKey instead.", DiagnosticId = "SYSLIB0033", UrlFormat = "https://aka.ms/dotnet-warnings/{0}")]
```

**Review by jeffhandley — APPROVED** [review]

**Comment by jeffhandley** [comment]
Breaking change document issue created: dotnet/docs#25506

**Review by bartonjs — APPROVED** [review]

**Pull request merged**